File: src/jump/sudoku/sudoku_test.jl
import DotEnv
DotEnv.config()

using MOSInterface

@testset "Sudoku" begin

    # Interface
    interface = Interface()

    # Delete model
    delete_model_with_name(interface, "Sudoku JuMP Model")

    # New model
    model = new_model(interface, "./src/jump/sudoku/sudoku_model.jl")

    # Get model
    model = get_model_with_name(interface, "Sudoku JuMP Model")

    # Set inputs
    set_interface_file(model, "initial_grid_file", "./src/jump/sudoku/data.csv")

    # Show info
    @test get_name(model) == "Sudoku JuMP Model"
    @test get_system(model) == "jump"
    @test get_status(model) == "created"

    # Solve
    MOSInterface.run(model)

    # Status
    @test get_status(model) == "success"

end

File: src/jump/sudoku/sudoku_test.py
from dotenv import load_dotenv, find_dotenv
load_dotenv(find_dotenv())

from mos.interface import Interface

def test_sudoku():

    # Interface
    interface = Interface()

    # Delte model
    interface.delete_model_with_name('Sudoku JuMP Model')

    # New model
    model = interface.new_model('./src/jump/sudoku/sudoku_model.jl')

    # Existing model by name
    model = interface.get_model_with_name('Sudoku JuMP Model')

    # Set inputs
    model.set_interface_file('initial_grid_file', './src/jump/sudoku/data.csv')

    # Solve 
    model.run()

    assert(model.get_status() == 'success')

if __name__ == '__main__':
    test_sudoku()

